fulltext/process/tests/test_process_psv.py

```diff
@@ -0,0 +1,147 @@
+        """The text has symbols."""
+        raw = "Bacon ipsum$@@ dolor amet lan!!!#djaeger chuc&&&^k bacon"
+        expected = "Bacon ipsum    dolor amet lan    djaeger chuc    k bacon"
+        self.assertEqual(psv._remove_Symbols(raw), expected,
+                         "Symbols should be removed")
+
+    def test_text_has_numbers(self):
+        """The text has numbers."""
+        raw = "Pork 2chop boudin5 picanha chic4ken"
+        expected = "Pork  chop boudin  picanha chic ken"
+        self.assertEqual(psv._remove_Numbers(raw), expected,
+                         "Numbers should be removed")
+
+    def test_text_has_unwanted_keywords(self):
+        """The text has some unwanted content identified by keywords."""
+        raw = [
+            "Bacon ipsum dolor amet landjaeger chuck bacon boudin sausage",
+            "arxiv ribs meatloaf chicken turducken bresaola shoulder. Pork",
+            "chop boudin will be inserted by hand later picanha chicken short",
+            "loin alcatra, turducken flank t-bone tail sirloin hamburger",
+            "turkey short ribs prosciutto. Pork was prepared with the aas",
+            "chop ribeye strip steak jerky, ball tip andouille leberkas cupim",
+            "1234567890",
+            "university",
+            "ham. Pig meatloaf short ribs leberkas, cupim pork chop",
+        ]
+        expected = [
+            "Bacon ipsum dolor amet landjaeger chuck bacon boudin sausage",
+            "loin alcatra, turducken flank t-bone tail sirloin hamburger",
+            "chop ribeye strip steak jerky, ball tip andouille leberkas cupim",
+            "1234567890",
+            "ham. Pig meatloaf short ribs leberkas, cupim pork chop",
+        ]
+        self.assertEqual(expected, psv._remove_Keyword(raw))
+
+    def test_text_has_continuations(self):
+        """The text has hyphenated continuations. Yuck."""
+        raw = [
+            "Bacon ipsum dolor amet landjaeger chuck bacon boudin saus- ",
+            "age.",
+            "Chop boudin picanha chicken short ",
+            "hmmm"
+        ]
+        expected = [
+            "",
+            "Bacon ipsum dolor amet landjaeger chuck bacon boudin sausage.",
+            "Chop boudin picanha chicken short hmmm"
+        ]
+        self.assertEqual(expected, psv._remove_BadEOL(raw))
+
+    def test_whitespace_all_over_the_place(self):
+        """The text has a variety of whitespace."""
+        raw = [
+            "Meatball\t pastrami chicken hamburger brisket ham hock capicola.",
+            "Shankle turkey tongue\n\nsirloin meatloaf corned beef tail strip",
+            "steak   sausage bacon beef ribs. "
+        ]
+        expected = [
+            "Meatball  pastrami chicken hamburger brisket ham hock capicola.",
+            "Shankle turkey tongue  sirloin meatloaf corned beef tail strip",
+            "steak   sausage bacon beef ribs. "
+        ]
+        self.assertEqual(expected, psv._remove_WhiteSpace(raw))
```

Comment:
You might want a test for idempotence. While it would only catch rare bugs, it would be super quick to write.

Reply:
This look something like what you had in mind?

```python
def test_remove_whitespace_is_idempotent(self):
        """Re-applying whitespace removal has no effect."""
        raw = [
            "Meatball\t pastrami chicken hamburger brisket ham hock capicola.",
            "Shankle turkey tongue\n\nsirloin meatloaf corned beef tail strip",
            "steak   sausage bacon beef ribs. "
        ]
        expected = [
            "Meatball  pastrami chicken hamburger brisket ham hock capicola.",
            "Shankle turkey tongue  sirloin meatloaf corned beef tail strip",
            "steak   sausage bacon beef ribs. "
        ]
        self.assertEqual(expected, psv._remove_WhiteSpace(raw))
        result = psv._remove_WhiteSpace(raw)
        for _ in range(5):
            result = psv._remove_WhiteSpace(result)
            self.assertEqual(expected, result)
```